### PR TITLE
change to use setuptools_scm for versioning

### PR DESCRIPTION
**Pull request by bhazelton**
This follows the approach we use in RASG repos.

### CODE REVIEW

**Comment by bhazelton** [comment]
@mkolopanis can you take a look at this?

File: setup.py [hunk 1]
```diff
@@ -31,13 +25,13 @@
...
     'author_email': "hera-sw@lists.berkeley.edu",
-    'version': version.version,
+    "use_scm_version": {"local_scheme": branch_scheme},
     'packages': ['hera_mc', 'hera_mc.tests'],
     'scripts': glob.glob('scripts/*'),
     'include_package_data': True,
     'install_requires': ["numpy", "astropy", "sqlalchemy", "psycopg2",
                          "alembic", "python-dateutil", "tabulate", "pandas",
-                         "psutil", "cartopy", "pyyaml", "redis"],
+                         "psutil", "cartopy", "pyyaml", "redis", "setuptools_scm"],
```
Comment (mkolopanis) [inline setup.py:34]: Do we actually need `setuptools_scm` as an install dependency? I know it is a build dependency and we have that declared in the `pyproject.toml`, but I wonder if it is necessary here or not.

**Review by mkolopanis — APPROVED** [review]
This looks good. Clean cut just like the RASG repos. I have one very small comment that doesn't really matter so I'll just go ahead and approve it.

File: setup.py (hunk 1 above)
Comment (bhazelton) [reply setup.py:34]: I think we need it as an install dependency because it's imported in the `__init__`.py.

Comment (mkolopanis) [reply setup.py:34]: fair enough! :D let's get this done.

**Pull request merged**